src/distribution/cauchy.rs

```diff
@@ -0,0 +1,238 @@
+
+    macro_rules! new(
+        ($loc:expr, $gamma:expr) => (Cauchy::new($loc, $gamma));
+    );
+
+    #[test]
+    fn density() {
+        let d = new!(2.0, 8.0);
+        let x = vec![-1.0, 0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 4.0, 6.0, 12.0];
+        let p = vec![
+            0.03488327519822364,
+            0.03744822190397538,
+            0.03843742021842001,
+            0.039176601376466544,
+            0.03963391578942141,
+            0.039788735772973836,
+            0.03963391578942141,
+            0.039176601376466544,
+            0.03744822190397538,
+            0.03183098861837907,
+            0.015527311521160521,
+        ];
+
+        assert::close(
+            &x.iter().map(|&x| d.density(x)).collect::<Vec<_>>(),
+            &p,
+            1e-15,
+        );
+    }
+
+    #[test]
+    fn distribution() {
+        let d = new!(2.0, 8.0);
+        let x = vec![
+            -1.0, 0.0, 0.01, 0.05, 0.1, 0.15, 0.25, 0.5, 1.0, 1.5, 2.0, 3.0, 4.0,
+        ];
+        let p = vec![
+            0.3857997487800918,
+            0.4220208696226307,
+            0.4223954618429798,
+            0.4238960166273086,
+            0.4257765641957529,
+            0.42766240385764065,
+            0.43144951512041,
+            0.44100191513247144,
+            0.46041657583943446,
+            0.48013147569445913,
+            0.5,
+            0.5395834241605656,
+            0.5779791303773694,
+        ];
+
+        assert::close(
+            &x.iter().map(|&x| d.distribution(x)).collect::<Vec<_>>(),
+            &p,
+            1e-15,
+        );
+    }
+
+    #[test]
+    fn entropy() {
+        use std::f64::consts::PI;
+        assert_eq!(new!(2.0, 1.0).entropy(), (PI * 4.0).ln());
+        assert::close(new!(3.0, 5.2).entropy(), 4.1796828725566719243, 1e-15);
+    }
+
+    #[test]
+    fn inverse() {
+        let d = new!(2.0, 3.0);
+        let x = vec![
+            -7.2330506115257585,
+            -0.9999999999999996,
+            2.0,
+            5.0,
+            11.233050611525758,
+        ];
+        let p = vec![0.1, 0.25, 0.5, 0.75, 0.9];
+
+        assert::close(
+            &p.iter().map(|&p| d.inverse(p)).collect::<Vec<_>>(),
+            &x,
+            1e-14,
+        );
+
+        assert!(d.inverse(0.0) < -1e16);
+        assert!(d.inverse(1.0) > 1e16);
+    }
+
+    #[test]
+    fn median() {
+        assert_eq!(new!(2.0, 1.0).median(), 2.0);
+    }
+
+    #[test]
+    fn modes() {
+        assert_eq!(new!(2.0, 1.0).modes(), vec![2.0]);
+    }
+
+    #[test]
+    fn sampling() {
+        let n = 100000;
+        let d = Cauchy::new(35.4, 12.3);
+        let mut source = source::Xorshift128Plus::new([42, 69]);
+
+        // Estimate the Kullback-Leibler divergence `KL(samples || d)` based on
+        // `n` samples and assert that the estimate is close to zero.
+        // This test is reproducible because we use a fixed random seed.
+        let cross_entropy_estimate = -(0..n)
```

Comment:
Just call it `entropy`.

Reply:
This is the cross entropy, not the entropy. But I can drop the `_estimate` suffix. I think it would be confusing to call this `entropy` because we later calculate the KL-divergence, which is cross entropy minus entropy. If we called this `entropy` then the KL-divergence below would become "entropy minus entropy", which is weird.